buildtest/buildsystem/base.py

```diff
@@ -466,31 +519,82 @@ def run_tests(self, testfile):
         # Keep an output file
         run_output_file = os.path.join(self.metadata.get("rundir"), self.build_id)
 
+        self.metadata["outfile"] = run_output_file + ".out"
+        self.metadata["errfile"] = run_output_file + ".err"
+
         # write output of test to .out file
-        with open(run_output_file + ".out", "w") as fd:
-            fd.write("\n".join(out))
-        self.logger.debug(f"Writing run output to file: {run_output_file+'.out'}")
+
+        out = "\n".join(out)
+        err = "\n".join(err)
+
+        write_file(self.metadata["outfile"], out)
+        self.logger.debug(f"Writing run output to file: {self.metadata['outfile']}")
+
         # write error from test to .err file
-        with open(run_output_file + ".err", "w") as fd:
-            fd.write("\n".join(err))
-        self.logger.debug(f"Writing run error to file: {run_output_file + '.err'}")
+        write_file(self.metadata["errfile"], err)
+        self.logger.debug(f"Writing run error to file: {self.metadata['errfile']}")
+
         self.logger.debug(f"Return code: {command.returncode} for test: {testfile}")
         result["RETURN_CODE"] = command.returncode
         result["END_TIME"] = self.get_formatted_time("end_time")
 
-        # Print the test result for the user
-        if command.returncode == 0:
-            print(
-                "{:<30} {:<30} {:<30} {:<30}".format(
-                    self.config_name, self.name, "PASSED", self.buildspec
+        status = self.recipe.get("status")
+
+        test_state = ""
```

Comment:
Any reason to not have it start as FAIL up here, and then just update to PASS given passing conditions? Right now there is an if -> else flow, and the else always sets it to FAIL - why not just start like that? A default state is to FAIL until there is evidence of passing. That will also simplify some of the logic below.

Reply:
good call, I see your point, set **test_state="FAIL"** and remove the else conditions. Ill fix this.